compiler/modules/front/symbols.bal

```diff
@@ -11,6 +12,21 @@ type ModuleSymbols record {|
     map<Import>[] partPrefixes = [];
     t:Context tc;
     boolean allowAllTypes = false;
+    ResolvedType[] deferredEmptinessChecks = [];
+    table<AtomicEmptyType> key(startPos, endPos) atomicEmptyTypeRecursionCheck = table [];
+|};
+
+type ResolvedType record {|
+    t:SemType semType;
+    s:ModuleLevelDefn modDefn;
+    s:TypeDesc td;
+|};
+
+type AtomicEmptyType record {|
+    readonly d:Position startPos;
```

Comment:
Using startPos and endPos doesn't uniquely identify the type desc (because there can be multiple source parts with the same position).

You can use partIndex in the identifier and SourceFile to construct the location.

But I'm not sure you need this table.

Reply:
removed table